controllers/loadtest_controller.go

```diff
@@ -139,12 +155,38 @@ func (r *LoadTestReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		}
 	}
 
+	reQueueTime := getRequeueTime(test, previousStatus, log)
+	if reQueueTime != 0 {
+		return ctrl.Result{RequeueAfter: reQueueTime}, nil
+	}
 	return ctrl.Result{}, nil
 }
 
-// LoadTestMissing categorize missing components based on their roles at specific
-// moment. The struct is a wrapper to help us get role information associate
-// with components.
+// getRequeueTime takes a LoadTest and its previous state, and compares the
+// previous status of the load test with its updated status and return a
+// calculated requeue time. The reason we take the test itself other than
+// its current status is because the ttl and time out are saved in its spec.
+// if the load test has just been assigned a start time, getRequeueTime returns
+// the timeout specified within its spec. If the load test has been just
+// assigned stop time, getRequeueTime returns its ttl specified within its spec
+// less its actual running time. In other cases, returns a 0 value time.duration
```

Comment:
nit: I think @paulosjca has reviewed me on spelling out small numbers before. He may want this to be changed from 0 to zero.

Reply:
I remembered that :) Will fix them.